M2/Macaulay2/packages/KGAlgebras.m2

```diff
@@ -0,0 +1,192 @@
+                Mp=Mp*(apply((entries (coefficients P)#1),x-> x#0))#m;
+                MP = MP+Mp
+            );
+            
+
+            --MP is p(theta) in the algorithm
+            N := kernel MP;
+            y := (gens N)_0;--Takes a "random" element of the kernel
+            M1 := matrix apply(group G, z-> z*y); --multiplies all group elements by the random element, giving column vectors, and we make a matrix M1 from those vectors
+            if rank M1 != rank ((group G)#0) then (
+                return image M1;
+            );--Checks that the module generated by M1 is not a strict submodule of KG
+            
+            H := apply(group G, transpose); --Generate list of group elements' matrices but transposed
+            MPP := transpose MP;
+            NP := kernel MPP;
+            yp := (gens NP)_0;
+            M2 := matrix apply(H,z -> z*yp);
+            if debugLevel == 1 then (
+                print(rank M2);
+            );
+            if rank M2 != rank (H#0) then (
+                return image M2;
+            );
+            if dim N == (degree(P))#0 then (
+                                return true;
+                        );
+                        
+        );
+    );
+);
+
+-- symmetric groups
+SnGroup = method(Options => {"K" => QQ})
+SnGroup (ZZ) := opts -> (n) -> (
+    xx := symbol xx;
+    R := (opts#"K")[xx_1 .. xx_n]; -- polynomial ring to do the group action on
+    cycle := new Array from 1 .. n; -- an n-cycle
+        finiteAction({permutationMatrix(n, cycle), -- the cycle
+            permutationMatrix(n, [1,2])}, -- a transposition
+            R)
+);
+
+
+
+
+
+
+beginDocumentation()
+
+doc ///
+    Key
+        KGAlgebras
+    Headline
+        A package for group rings
+    Description
+        Text
+            Implements group rings and the meataxe algorithm to compute the irreducibility of a finite dimensional representation over a finite field.
+
+///
+doc ///
+    Key
+        groupRing
+    Headline
+        Constructor for GroupRing
+    Usage
+        groupRing(G,K)
+    Inputs
+        G:
+            A FiniteGroupAction specifying a group.
+        K:
+            A field. 
+    Outputs
+        :
+            A group ring defined by group G and field K. 
+    Description
+        Text
+            Takes a FiniteGroupAction G and a field K and creates a group ring from them. 
+        Example
+            R = QQ[x1,x2,x3];
+            L = {matrix {{0,1,0},{1,0,0},{0,0,1}}, matrix {{0,0,1},{0,1,0},{1,0,0}}, matrix {{1,0,0},{0,0,1},{0,1,0}} };
+            G = finiteAction(L,R);
+            groupRing(G,QQ)
+///
+doc ///
+    Key
+        meataxe        
+    Headline
+        Boolean function implementing the meataxe algorithm determining the irreducibility of a finite group representation of a finite dimensional vector space. If the module is irreducible, returns true. Otherwise, the method returns a nontrivial submodule. 
```

Comment:
The builds are failing with the following error:

```m2
../../../../../Macaulay2/m2/document.m2:598:32:(1):[72]: error: document: documentation headlines must be less than 100 characters long:
  "Boolean function implementing the meataxe algorithm determining the irreducibility of a finite group representation of a finite dimensional vector space. If the module is irreducible, returns true. Otherwise, the method returns a nontrivial submodule."
```
In general, headlines should just be a few words.  Most of this stuff can get moved elsewhere in the documentation node.